File: app/web/static/js/integrations.js
// Integrations Manager - Full Implementation
// Conecta o frontend com as rotas de API

let voiceEnabled = false;
let voiceRecognition = null;

// ===== INICIALIZAÇÃO =====

document.addEventListener("DOMContentLoaded", () => {
  loadIntegrationsStatus();
  setupEventListeners();
  initVoiceRecognition();
});

function setupEventListeners() {
  // Discord
  document
    .getElementById("btn-save-discord")
    ?.addEventListener("click", saveDiscord);
  document
    .getElementById("btn-test-discord")
    ?.addEventListener("click", testDiscord);

  // Minecraft
  document
    .getElementById("btn-save-minecraft")
    ?.addEventListener("click", saveMinecraft);
  document
    .getElementById("btn-test-minecraft")
    ?.addEventListener("click", testMinecraft);

  // Email
  document
    .getElementById("btn-save-email")
    ?.addEventListener("click", saveEmail);
  document
    .getElementById("btn-test-email")
    ?.addEventListener("click", testEmail);

  // Twitch API
  document
    .getElementById("btn-save-twitch-api")
    ?.addEventListener("click", saveTwitchAPI);

  // Import
  document
    .getElementById("btn-import-streamelements")
    ?.addEventListener("click", () => importFile("streamelements"));
  document
    .getElementById("btn-import-nightbot")
    ?.addEventListener("click", () => importFile("nightbot"));
}

// ===== CARREGAR STATUS =====

async function loadIntegrationsStatus() {
  try {
    const res = await fetch("/api/integrations/status");
    const data = await res.json();

    console.log("📊 Status das integrações:", data);

    // Atualizar badges de status
    updateStatusBadge("discord", data.discord);
    updateStatusBadge("minecraft", data.minecraft);
    updateStatusBadge("email", data.email);
    updateStatusBadge("twitch-api", data.twitch_api);
  } catch (err) {
    console.error("Erro ao carregar status:", err);
    showToast("Erro ao carregar status das integrações", "error");
  }
}

function updateStatusBadge(name, status) {
  const badge = document.getElementById(`status-${name}`);
  if (!badge) return;

  if (status.enabled) {
    badge.textContent = "🟢 Ativado";
    badge.className = "bg-green-600 text-xs px-3 py-1 rounded-full";
  } else {
    badge.textContent = "🔴 Desativado";
    badge.className = "bg-red-600 text-xs px-3 py-1 rounded-full";
  }

  // Atualizar disponibilidade
  if (!status.available) {
    const card = document.getElementById(`card-${name}`);
    if (card) {
      const warning = document.createElement("div");
      warning.className =
        "bg-yellow-600/20 border border-yellow-600 rounded-lg p-2 mt-2 text-xs";
      warning.textContent = "⚠️ Biblioteca não instalada";
      card.querySelector(".space-y-3")?.after(warning);
    }
  }
}

// ===== DISCORD =====

async function saveDiscord() {
  const token = document.getElementById("discord-token").value.trim();
  const channelId = document.getElementById("discord-channel").value.trim();

  if (!token || !channelId) {
    showToast("Preencha o Token e o Channel ID", "warning");
    return;
  }

  try {
    const res = await fetch("/api/integrations/discord/setup", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ token, channel_id: channelId }),
    });

    const data = await res.json();

    if (res.ok) {
      showToast("✅ Discord configurado com sucesso!", "success");
      loadIntegrationsStatus();
    } else {
      showToast(`❌ ${data.error}`, "error");
    }
  } catch (err) {
    console.error("Erro ao salvar Discord:", err);
    showToast("Erro de conexão", "error");
  }
}

async function testDiscord() {
  try {
    const res = await fetch("/api/integrations/discord/test", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        message: "🎮 Teste de integração do Twitch Bot!",
      }),
    });

    const data = await res.json();
    showToast(
      res.ok ? `✅ ${data.message}` : `❌ ${data.error}`,
      res.ok ? "success" : "error"
    );
  } catch (err) {
    showToast("Erro ao testar Discord", "error");
  }
}

async function disableDiscord() {
  try {
    const res = await fetch("/api/integrations/discord/disable", {
      method: "POST",
    });
    const data = await res.json();

    if (res.ok) {
      showToast("Discord desabilitado", "info");
      loadIntegrationsStatus();
    }
  } catch (err) {
    showToast("Erro ao desabilitar", "error");
  }
}

// ===== MINECRAFT =====

async function saveMinecraft() {
  const host = document.getElementById("minecraft-host").value.trim();
  const port = parseInt(document.getElementById("minecraft-port").value);
  const password = document.getElementById("minecraft-password").value.trim();

  if (!host || !password) {
    showToast("Preencha o Host e a Senha RCON", "warning");
    return;
  }

  try {
    const res = await fetch("/api/integrations/minecraft/setup", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ host, port, password }),
    });

    const data = await res.json();

    if (res.ok) {
      showToast("✅ Minecraft configurado com sucesso!", "success");
      loadIntegrationsStatus();
    } else {
      showToast(`❌ ${data.error}`, "error");
    }
  } catch (err) {
    console.error("Erro ao salvar Minecraft:", err);
    showToast("Erro de conexão", "error");
  }
}

async function testMinecraft() {
  try {
    const res = await fetch("/api/integrations/minecraft/test", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        command: "say Teste de integração com a Twitch!",
      }),
    });

    const data = await res.json();
    showToast(
      res.ok ? `✅ ${data.message}` : `❌ ${data.error}`,
      res.ok ? "success" : "error"
    );
  } catch (err) {
    showToast("Erro ao testar Minecraft", "error");
  }
}

// ===== EMAIL =====

async function saveEmail() {
  const smtpServer = document.getElementById("email-smtp").value.trim();
  const port = parseInt(document.getElementById("email-port").value);
  const email = document.getElementById("email-address").value.trim();
  const password = document.getElementById("email-password").value.trim();

  if (!email || !password) {
    showToast("Preencha o Email e a Senha", "warning");
    return;
  }

  try {
    const res = await fetch("/api/integrations/email/setup", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ smtp_server: smtpServer, port, email, password }),
    });

    const data = await res.json();

    if (res.ok) {
      showToast("✅ Email configurado com sucesso!", "success");
      loadIntegrationsStatus();
    } else {
      showToast(`❌ ${data.error}`, "error");
    }
  } catch (err) {
    console.error("Erro ao salvar Email:", err);
    showToast("Erro de conexão", "error");
  }
}

async function testEmail() {
  try {
    const res = await fetch("/api/integrations/email/test", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        subject: "Teste de Integração - Twitch Bot",
        body: "Este é um email de teste da integração com o Twitch Bot! 🎮",
      }),
    });

    const data = await res.json();
    showToast(
      res.ok ? `✅ ${data.message}` : `❌ ${data.error}`,
      res.ok ? "success" : "error"
    );
  } catch (err) {
    showToast("Erro ao testar Email", "error");
  }
}

// ===== TWITCH API =====

async function saveTwitchAPI() {
  const clientId = document.getElementById("twitch-client-id").value.trim();
  const clientSecret = document
    .getElementById("twitch-client-secret")
    .value.trim();

  if (!clientId || !clientSecret) {
    showToast("Preencha o Client ID e Client Secret", "warning");
    return;
  }

  try {
    const res = await fetch("/api/integrations/twitch-api/setup", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({
        client_id: clientId,
        client_secret: clientSecret,
      }),
    });

    const data = await res.json();

    if (res.ok) {
      showToast("✅ Twitch API configurada com sucesso!", "success");
      loadIntegrationsStatus();
    } else {
      showToast(`❌ ${data.error}`, "error");
    }
  } catch (err) {
    console.error("Erro ao salvar Twitch API:", err);
    showToast("Erro de conexão", "error");
  }
}

// ===== RECONHECIMENTO DE VOZ =====

function initVoiceRecognition() {
  if (
    !("webkitSpeechRecognition" in window) &&
    !("SpeechRecognition" in window)
  ) {
    console.warn("⚠️ Navegador não suporta reconhecimento de voz");
    return;
  }

  const SpeechRecognition =
    window.SpeechRecognition || window.webkitSpeechRecognition;
  voiceRecognition = new SpeechRecognition();

  voiceRecognition.continuous = true;
  voiceRecognition.interimResults = false;
  voiceRecognition.lang = "pt-BR";

  voiceRecognition.onresult = (event) => {
    const last = event.results.length - 1;
    const text = event.results[last][0].transcript.toLowerCase();
    console.log("🎤 Reconhecido:", text);

    if (text.includes("bot")) {
      processVoiceCommand(text);
    }
  };

  voiceRecognition.onerror = (event) => {
    console.error("Erro no reconhecimento:", event.error);
  };
}

function toggleVoice() {
  voiceEnabled = !voiceEnabled;
  const btn = document.getElementById("toggle-voice");
  const status = document.getElementById("voice-status");

  if (voiceEnabled) {
    if (voiceRecognition) {
      voiceRecognition.start();
      btn.textContent = "🔴 Desativar Reconhecimento";
      btn.className =
        "w-full bg-green-600 hover:bg-green-700 py-2 rounded-lg transition";
      status.textContent = "🟢 Escutando";
      status.className = "bg-green-600 text-xs px-3 py-1 rounded-full";
      showToast("Reconhecimento de voz ativado!", "success");
    } else {
      showToast("Reconhecimento de voz não disponível", "error");
      voiceEnabled = false;
    }
  } else {
    if (voiceRecognition) {
      voiceRecognition.stop();
    }
    btn.textContent = "🎤 Ativar Reconhecimento";
    btn.className =
      "w-full bg-red-600 hover:bg-red-700 py-2 rounded-lg transition";
    status.textContent = "🔴 Desativado";
    status.className = "bg-red-600 text-xs px-3 py-1 rounded-full";
    showToast("Reconhecimento de voz desativado", "info");
  }
}

function processVoiceCommand(text) {
  console.log("🎯 Processando comando:", text);
  // Comandos já implementados no dashboard.js
  showToast(`Comando reconhecido: ${text}`, "info");
}

// ===== IMPORTAÇÃO DE ARQUIVOS =====

function importFile(type) {
  const input = document.createElement("input");
  input.type = "file";
  input.accept = type === "streamelements" ? ".csv" : ".json";

  input.onchange = async (e) => {
    const file = e.target.files[0];
    if (!file) return;

    const formData = new FormData();
    formData.append("file", file);

    const endpoint =
      type === "streamelements"
        ? "/api/import/streamelements"
        : "/api/import/nightbot";

    try {
      showToast("⏳ Importando...", "info");

      const res = await fetch(endpoint, {
        method: "POST",
        body: formData,
      });

      const data = await res.json();

      if (res.ok) {
        showToast(
          `✅ ${data.imported} itens importados com sucesso!`,
          "success"
        );
      } else {
        showToast(`❌ ${data.error}`, "error");
      }
    } catch (err) {
      console.error("Erro ao importar:", err);
      showToast("Erro ao importar arquivo", "error");
    }
  };

  input.click();
}

// ===== TOAST NOTIFICATIONS =====

function showToast(message, type = "info") {
  const colors = {
    success: "bg-green-600",
    error: "bg-red-600",
    warning: "bg-yellow-600",
    info: "bg-blue-600",
  };

  const toast = document.createElement("div");
  toast.className = `fixed bottom-8 right-8 ${colors[type]} text-white px-6 py-3 rounded-lg shadow-lg z-50`;
  toast.textContent = message;

  document.body.appendChild(toast);

  setTimeout(() => {
    toast.style.transition = "opacity 0.3s";
    toast.style.opacity = "0";
    setTimeout(() => toast.remove(), 300);
  }, 3000);
}

// Exportar funções globais
window.toggleVoice = toggleVoice;
window.saveDiscord = saveDiscord;
window.testDiscord = testDiscord;
window.saveMinecraft = saveMinecraft;
window.testMinecraft = testMinecraft;
window.saveEmail = saveEmail;
window.testEmail = testEmail;
window.saveTwitchAPI = saveTwitchAPI;

File: app/web/routes/integrations.py
"""
Rotas de API para Integrações
Discord, Minecraft, Email, Twitch API
"""

from flask import Blueprint, request, jsonify
from app.web.app_state import integration_manager

integrations_bp = Blueprint("integrations", __name__, url_prefix="/api/integrations")

# ===== STATUS DAS INTEGRAÇÕES =====

@integrations_bp.route("/status", methods=["GET"])
def get_integrations_status():
    """Retorna status de todas as integrações"""
    try:
        config = integration_manager.config
        dependencies = integration_manager.check_integration_dependencies() if hasattr(integration_manager, 'check_integration_dependencies') else {}
        
        return jsonify({
            "discord": {
                "enabled": config.get("discord", {}).get("enabled", False),
                "configured": bool(config.get("discord", {}).get("token")),
                "available": dependencies.get("Discord", False)
            },
            "minecraft": {
                "enabled": config.get("minecraft", {}).get("enabled", False),
                "configured": bool(config.get("minecraft", {}).get("host")),
                "available": dependencies.get("Minecraft", False)
            },
            "email": {
                "enabled": config.get("email", {}).get("enabled", False),
                "configured": bool(config.get("email", {}).get("email")),
                "available": dependencies.get("Email", False)
            },
            "twitch_api": {
                "enabled": config.get("twitch_api", {}).get("enabled", False),
                "configured": bool(config.get("twitch_api", {}).get("client_id")),
                "available": dependencies.get("Requests", False)
            }
        })
    except Exception as e:
        return jsonify({"error": str(e)}), 500


# ===== DISCORD =====

@integrations_bp.route("/discord/setup", methods=["POST"])
def setup_discord():
    """Configura integração com Discord"""
    try:
        data = request.json
        token = data.get("token", "").strip()
        channel_id = data.get("channel_id", "").strip()
        
        if not token or not channel_id:
            return jsonify({"error": "Token e Channel ID são obrigatórios"}), 400
        
        success, message = integration_manager.setup_discord(token, channel_id)
        
        if success:
            return jsonify({"status": "success", "message": message})
        else:
            return jsonify({"error": message}), 400
            
    except Exception as e:
        return jsonify({"error": str(e)}), 500


@integrations_bp.route("/discord/test", methods=["POST"])
def test_discord():
    """Testa envio de mensagem para Discord"""
    try:
        data = request.json
        message = data.get("message", "Teste de integração do Twitch Bot!")
        
        success, response = integration_manager.send_to_discord(message)
        
        if success:
            return jsonify({"status": "success", "message": response})
        else:
            return jsonify({"error": response}), 400
            
    except Exception as e:
        return jsonify({"error": str(e)}), 500


@integrations_bp.route("/discord/disable", methods=["POST"])
def disable_discord():
    """Desabilita integração com Discord"""
    try:
        integration_manager.config["discord"]["enabled"] = False
        integration_manager.save_config()
        return jsonify({"status": "success", "message": "Discord desabilitado"})
    except Exception as e:
        return jsonify({"error": str(e)}), 500


# ===== MINECRAFT =====

@integrations_bp.route("/minecraft/setup", methods=["POST"])
def setup_minecraft():
    """Configura integração com Minecraft (RCON)"""
    try:
        data = request.json
        host = data.get("host", "localhost").strip()
        port = int(data.get("port", 25575))
        password = data.get("password", "").strip()
        
        if not host or not password:
            return jsonify({"error": "Host e senha são obrigatórios"}), 400
        
        success, message = integration_manager.setup_minecraft(host, port, password)
        
        if success:
            return jsonify({"status": "success", "message": message})
        else:
            return jsonify({"error": message}), 400
            
    except Exception as e:
        return jsonify({"error": str(e)}), 500


@integrations_bp.route("/minecraft/test", methods=["POST"])
def test_minecraft():
    """Testa comando no servidor Minecraft"""
    try:
        data = request.json
        command = data.get("command", "say Teste de integração!")
        
        success, response = integration_manager.send_to_minecraft(command)
        
        if success:
            return jsonify({"status": "success", "message": response})
        else:
            return jsonify({"error": response}), 400
            
    except Exception as e:
        return jsonify({"error": str(e)}), 500


@integrations_bp.route("/minecraft/disable", methods=["POST"])
def disable_minecraft():
    """Desabilita integração com Minecraft"""
    try:
        integration_manager.config["minecraft"]["enabled"] = False
        integration_manager.save_config()
        return jsonify({"status": "success", "message": "Minecraft desabilitado"})
    except Exception as e:
        return jsonify({"error": str(e)}), 500


# ===== EMAIL =====

@integrations_bp.route("/email/setup", methods=["POST"])
def setup_email():
    """Configura integração com Email"""
    try:
        data = request.json
        smtp_server = data.get("smtp_server", "smtp.gmail.com").strip()
        port = int(data.get("port", 587))
        email = data.get("email", "").strip()
        password = data.get("password", "").strip()
        
        if not email or not password:
            return jsonify({"error": "Email e senha são obrigatórios"}), 400
        
        success, message = integration_manager.setup_email(smtp_server, port, email, password)
        
        if success:
            return jsonify({"status": "success", "message": message})
        else:
            return jsonify({"error": message}), 400
            
    except Exception as e:
        return jsonify({"error": str(e)}), 500


@integrations_bp.route("/email/test", methods=["POST"])
def test_email():
    """Testa envio de email"""
    try:
        data = request.json
        to_email = data.get("to", integration_manager.config["email"]["email"])
        subject = data.get("subject", "Teste de Integração - Twitch Bot")
        body = data.get("body", "Este é um email de teste da integração do Twitch Bot!")
        
        success, response = integration_manager.send_email(to_email, subject, body)
        
        if success:
            return jsonify({"status": "success", "message": response})
        else:
            return jsonify({"error": response}), 400
            
    except Exception as e:
        return jsonify({"error": str(e)}), 500


@integrations_bp.route("/email/disable", methods=["POST"])
def disable_email():
    """Desabilita integração com Email"""
    try:
        integration_manager.config["email"]["enabled"] = False
        integration_manager.save_config()
        return jsonify({"status": "success", "message": "Email desabilitado"})
    except Exception as e:
        return jsonify({"error": str(e)}), 500


# ===== TWITCH API =====

@integrations_bp.route("/twitch-api/setup", methods=["POST"])
def setup_twitch_api():
    """Configura Twitch API avançada"""
    try:
        data = request.json
        client_id = data.get("client_id", "").strip()
        client_secret = data.get("client_secret", "").strip()
        
        if not client_id or not client_secret:
            return jsonify({"error": "Client ID e Client Secret são obrigatórios"}), 400
        
        success, message = integration_manager.setup_twitch_api(client_id, client_secret)
        
        if success:
            return jsonify({"status": "success", "message": message})
        else:
            return jsonify({"error": message}), 400
            
    except Exception as e:
        return jsonify({"error": str(e)}), 500


@integrations_bp.route("/twitch-api/user/<username>", methods=["GET"])
def get_twitch_user_id(username):
    """Obtém ID de um usuário da Twitch"""
    try:
        user_id = integration_manager.get_user_id(username)
        
        if user_id:
            return jsonify({"user_id": user_id, "username": username})
        else:
            return jsonify({"error": "Usuário não encontrado"}), 404
            
    except Exception as e:
        return jsonify({"error": str(e)}), 500


@integrations_bp.route("/twitch-api/followers/<broadcaster_id>", methods=["GET"])
def get_followers(broadcaster_id):
    """Obtém seguidores de um canal"""
    try:
        followers = integration_manager.get_channel_followers(broadcaster_id)
        return jsonify({"total": len(followers), "followers": followers})
    except Exception as e:
        return jsonify({"error": str(e)}), 500


@integrations_bp.route("/twitch-api/disable", methods=["POST"])
def disable_twitch_api():
    """Desabilita Twitch API"""
    try:
        integration_manager.config["twitch_api"]["enabled"] = False
        integration_manager.save_config()
        return jsonify({"status": "success", "message": "Twitch API desabilitada"})
    except Exception as e:
        return jsonify({"error": str(e)}), 500